widgets/src/lib/InferenceWidget/widgets/ZeroShotImageClassificationWidget/ZeroShotImageClassificationWidget.svelte

```diff
@@ -0,0 +1,159 @@
+<script>
+	import type { WidgetProps } from "../../shared/types";
+
+	import WidgetCheckbox from "../../shared/WidgetCheckbox/WidgetCheckbox.svelte";
+	import WidgetDropzone from "../../shared/WidgetDropzone/WidgetDropzone.svelte";
+	import WidgetOutputChart from "../../shared/WidgetOutputChart/WidgetOutputChart.svelte";
+	import WidgetSubmitBtn from "../../shared/WidgetSubmitBtn/WidgetSubmitBtn.svelte";
+	import WidgetTextInput from "../../shared/WidgetTextInput/WidgetTextInput.svelte";
+	import WidgetWrapper from "../../shared/WidgetWrapper/WidgetWrapper.svelte";
+	import { getResponse } from "../../shared/helpers";
+
+	export let apiToken: WidgetProps["apiToken"];
+	export let apiUrl: WidgetProps["apiUrl"];
+	export let model: WidgetProps["model"];
+	export let noTitle: WidgetProps["noTitle"];
+
+	const accept = "image/*";
+	let candidateLabels = "";
+	let computeTime = "";
+	let error: string = "";
+	let isLoading = false;
+	let modelLoading = {
+		isLoading: false,
+		estimatedTime: 0,
+	};
+	let multiClass = false;
+	let output: Array<{ label: string; score: number }> = [];
+	let outputJson: string;
+	let imageBase64 = "";
+
+	function onSelectFile(file: File) {
+		error = file.type.match(accept) ? "" : "You need to upload an image";
+		if (error) {
+			imageBase64 = "";
+			return;
+		}
+
+		let fileReader: FileReader = new FileReader();
+		fileReader.onload = () => {
+			const imageBase64WithPrefix: string = fileReader.result as string;
+			imageBase64 = imageBase64WithPrefix.split(",")[1]; // remove prefix
+			isLoading = false;
+		};
+		isLoading = true;
+		fileReader.readAsDataURL(file);
+	}
+
+	async function getOutput(withModelLoading = false) {
+		const trimmedCandidateLabels = candidateLabels.trim().split(",").join(",");
+
+		if (!imageBase64) {
+			error = "You need to upload an image";
+			output = [];
+			outputJson = "";
+			return;
+		}
+
+		if (!trimmedCandidateLabels) {
+			error = "You need to input at least one label";
+			output = [];
+			outputJson = "";
+			return;
+		}
+
+		const requestBody = {
+			parameters: {
+				images: [imageBase64],
```

Comment:
@osanseviero updated `requestBody` as suggested

Reply:
@Narsil could you take a look at the requestBody expectation? Should we also have multiClass for this pipeline? And does it make sense to have the `hypothesis_template` in the widget? 

(sorry @mishig25, `multiClass` and `hypothesis_template` should be two separate params)

```
const requestBody = {
	parameters: {
                images: [imageBase64],
		candidate_labels: trimmedCandidateLabels,
		hypothesisTemplate: hypothesisTemplate,
                multiClass: multiClass,
	},
};
```

Reply:
@osanseviero all good! What should be a value for `hypothesisTemplate`?

Reply:
Hi,

I don't think `candidate_labels` should be `parameters` tbh, it should be `inputs`.
For zero-shot-classification, it's `inputs: str, parameters: {candidate_labels: [str], multi_label: bool}` (but it's a bit confusing since `candidate_labels` is required whereas `multi_label` is not..., IMO it's more a real input of the query since we can't do anything without them. But BC so we probably won't change it for `text zero-shot`. Here getting it correct and using `{inputs: {image: base64, candidate_labels: [str], parameters: {multi_label: bool}}` seems a bit better (we could keep consistency too I'm fine with it)

I don't think we should support a list of images either (we don't use it for the widget anyway for starts). And batching is really a tricky thing within the API. (zero-shot is where it makes the most sense, but even there it's not perfect) anyhow, I think for the widget raw base64 string seems better IMO (maybe we need the content-type in order to load it, I'm not sure I would have to check how PIL works to loads raw bytes). 

`hypothesis_template` is nice in the API, but I am not sure it makes sense for the widget, most people use the default (in the API too actually). Not sending it over the wire is probably better if we can avoid. (Some people copy what's happening on the wire rather than reading docs, which is fine)

In the end I would personnally vote for :

```
{inputs: { image: base64str, candidate_labels: str[]}, parameters?: {multi_label: boo?}
```

Reply:
multi_label is actually interesting in some cases for text btw, to get "product" of classes.

like candidate_labels : ["medical", "legal", 'positive", "negative"]